### ISSUE TITLE
Allow to change the `anomaly.detection.goals` in cruise control

### ISSUE THREAD

**Issue by driosalido** [strimzi/strimzi-kafka-operator#5505]
**Is your feature request related to a problem? Please describe.**

The Cruise Control JMX Metric `anomalydetector_balancedness_score` is based on the`anomaly.detection.goals` goals, which by default are just: 

```
com.linkedin.kafka.cruisecontrol.analyzer.goals.RackAwareGoal,
com.linkedin.kafka.cruisecontrol.analyzer.goals.MinTopicLeadersPerBrokerGoal,
com.linkedin.kafka.cruisecontrol.analyzer.goals.ReplicaCapacityGoal,
com.linkedin.kafka.cruisecontrol.analyzer.goals.DiskCapacityGoal
``` 

so if those goals are all satisfied, then the anomalydetector_balancedness_score  is going to be 100

**Describe the solution you'd like**

Being able to redefine the `anomaly.detection.goals` to suite the `anomalydetector_balancedness_score` metric to your own priorities

**Describe alternatives you've considered**
Add a different `balancedness_score` metric that use the `goals` to determine the score

**Comment by tomncooper** [comment]
So the reason we don't allow the AD goals to be configured is that they are intrinsically linked to the self-healing functionality, which we don't currently support. CC does a series of checks at start up and originally the AD goals had to be a sub-set of the Self-healing goals and the default goals. As we didn't support the SH goals, we didn't want users to have to configure things we didn't support. 

However, in later version of CC (including the version in strimzi 0.25) this check seems to have been removed and AD goals no longer need to be a sub-set of the SH goals. So it we should be able to switch them to being configurable, but I will need to check how they play with self-healing features.

**Comment by driosalido** [comment]
That would be great Tom, having a configurable balancedness metric would be great for SRE and operations teams.

**Comment by kyguy** [comment]
> However, in later version of CC (including the version in strimzi 0.25) this check seems to have been removed and AD goals no longer need to be a sub-set of the SH goals. So it we should be able to switch them to being configurable, but I will need to check how they play with self-healing features.

@tomncooper  I was able to confirm locally that Cruise Control no longer requires`anomaly.detection.goals` to be a subset of SH goals or`default.goals`. However, Strimzi still has logic to keep `anomaly.detection.goals` as a subset of `default.goals`. This is a good thing though as without this logic it is possible to specify unsupported goals in the `anomaly.detector.goals` list.

Cruise Control ran smoothly with `anomaly.detection.goals` loaded with various goal configurations. From what I gather, these goals are strictly used for metrics/alerting and do not enable the self healing functionality of Cruise Control. Anyways, I have opened a PR and linked it to this issue!

Let me know what you think!
